### ISSUE TITLE
Incorrect socket selected in outbound mode

### ISSUE THREAD

**Issue by jonastelzio** [drachtio/drachtio-srf#117]
When running SRF in outbound connection mode, in functions like 'createB2BUA', the default socket selected is basically 'the first socket' it can find. The code workaround to this is forwarding the `_socket` parameter like this:

```
srf.createB2BUA(req, res, destSipUri, {
    headers: forwardHeaders,
    passFailure: false,
    localSdpB: rtpOffer.sdp,
    localSdpA: async (remoteSdp, res) => await this.getSdpAnswer(remoteSdp, res),
    _socket: req.socket  // <- pass on the incoming req.socket here
})
```

This workaround works, but it's interacting with a _prefixed parameter, and is completely undocumented.
If I don't set the _socket parameter, SRF will automatically just pick the first socket in the sockets collection, which can (and will in most cases), belong to a completely unrelated SRF session.

**Comment by davehorton** [comment]
Good point, would you agree that Srf.createB2BUA should set `_socket` for you?  If so I can make that change
